### PR TITLE
Add methods to safetly handle resources.

### PR DESCRIPTION
**Pull request by schmmd**

### CODE REVIEW

**Comment by schmmd** [comment]
This was my idea for safe resource methods re: our recent conversation.

File: core/src/main/scala/org/allenai/common/Resource.scala [hunk 1]
```diff
@@ -39,4 +41,47 @@ object Resource {
...
+    * @param  name  absolute path of the resource
+    */
+  def get(name: String): URL = {
+    getOpt(name).getOrElse {
+      throw new IllegalArgumentException("No such absolute resource found: " + name)
+    }
+  }
+
+  /** Get a Java Resource.
+    * This method provides a much nicer exception than the Java default (NPE).
```
Comment (jkinkead) [inline core/src/main/scala/org/allenai/common/Resource.scala]: This doesn't throw anything.

File: core/src/main/scala/org/allenai/common/Resource.scala [hunk 2]
```diff
@@ -39,4 +41,47 @@ object Resource {
...
+    *
+    * @param  clazz  the class from which the resource may be relative
+    * @returns  Some(URL) if the resource exists
+    */
+  def getOpt(clazz: Class[_], name: String): Option[URL] = {
```
Comment (jkinkead) [inline core/src/main/scala/org/allenai/common/Resource.scala]: I think the more accepted scala pattern is to do like:
```scala
import scala.reflect.{ classTag, ClassTag }
def get[T : ClassTag](name: String): URL = {
  Option(classTag[T].runtimeClass.getResource(name))
}
```

File: core/src/main/scala/org/allenai/common/Resource.scala [hunk 3]
```diff
@@ -39,4 +41,47 @@ object Resource {
...
+    getOpt(name).getOrElse {
+      throw new IllegalArgumentException("No such absolute resource found: " + name)
+    }
+  }
+
+  /** Get a Java Resource.
+    * This method provides a much nicer exception than the Java default (NPE).
+    *
+    * @param  name  absolute path of the resource
+    * @returns  Some(URL) if the resource exists
```
Comment (jkinkead) [inline core/src/main/scala/org/allenai/common/Resource.scala]: `@return`, not `@returns`. (. . . github tells me with its highlighting)

**Comment by jkinkead** [comment]
Awesome.

A couple random comments, LGTM!

**Pull request merged**